### PR TITLE
Константинов Илья. Вариант 20. Технология TBB. Поразрядная сортировка для вещественных чисел (тип double) с четно-нечетным слиянием Бэтчера.

### PR DESCRIPTION
**Pull request by Ilya2643**
Программа сортирует массив чисел с плавающей точкой (типа double) методом поразрядной сортировки. Процесс состоит из нескольких этапов:

Валидация: Проверяется, что входной и выходной массивы имеют одинаковый размер. Если размеры не совпадают, программа должна сообщить об ошибке и прекратить работу.
Препроцессинг (Предварительная обработка): Входные данные копируются во внутреннюю структуру данных программы (класс задачи). Это делается для удобства обработки и предотвращения случайных изменений исходных данных.
3 Обработка (Сортировка): Здесь происходит основная часть работы — собственно, поразрядная сортировка чисел с плавающей точкой. Поразрядная сортировка работает по принципу последовательного распределения элементов по "корзинам" в зависимости от значения отдельных битов (или частей) числа. Так как это числа с плавающей точкой, необходимо учитывать особенности их представления в памяти (мантиссу и экспоненту), что делает реализацию сложнее, чем для целых чисел.
Постпроцессинг (Последующая обработка): Отсортированные данные из внутренней структуры программы записываются в выходной массив, который был передан в функцию сортировки. Таким образом, пользователь получает отсортированный массив в том же месте, где он предоставил исходный массив.
В целом, алгоритм обеспечивает сортировку массива чисел с плавающей точкой, при этом гарантируя целостность входных данных и корректность выходного результата. Сложность реализации поразрядной сортировки для чисел с плавающей запятой обусловлена необходимостью обработки их внутреннего представления (мантисса и порядок).
Добавлена реализация распараллеливания за счет tbb в циклах с переводом в двоичное представление и обратно, а также параллельный подсчет счетчика байт для числа.

### CODE REVIEW

**Review by odincovm — CHANGES_REQUESTED** [review]
add a test to an empty array to test the functionality

**Review by ermilovad — CHANGES_REQUESTED** [review]
add a test with duplicates in the array

**Review by ermilovad — APPROVED** [review]

**Review by aobolensk — APPROVED** [review]

File: tasks/tbb/Konstantinov_I_Sort_Batcher/perf_tests/main.cpp [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+#include <memory>
+#include <random>
+#include <vector>
+
+#include "core/perf/include/perf.hpp"
+#include "core/task/include/task.hpp"
+#include "tbb/Konstantinov_I_Sort_Batcher/include/ops_tbb.hpp"
+
+TEST(Konstantinov_I_Sort_Batcher_tbb, test_pipeline_run) {
+  constexpr int kCount = 500000;
```
Comment (allnes) [inline tasks/tbb/Konstantinov_I_Sort_Batcher/perf_tests/main.cpp:16]: please use same data sizes for all vesrions
